barman/backup_executor.py

```diff
@@ -1388,6 +1384,286 @@ def _format_src(self, path):
+            reside.
+        :param list[str] snapshot_disks: A list containing the names of the disks for
+            which snapshots should be taken at backup time.
+        :rtype tuple[list[str],list[str]]
+        :return: A tuple where the first element is a list of all disks which are not
+            attached to the VM instance and the second element is a list of all disks
+            which are attached but not mounted.
+        """
+        attached_devices = snapshot_interface.get_attached_devices(
+            snapshot_instance, snapshot_zone
+        )
+        missing_disks = []
+        for disk in snapshot_disks:
+            if disk not in attached_devices.keys():
+                missing_disks.append(disk)
+
+        unmounted_disks = []
+        for disk in snapshot_disks:
+            try:
+                mount_point, _mount_options = cmd.findmnt(attached_devices[disk])
+            except KeyError:
+                # Ignore disks which were not attached
+                continue
+            if mount_point is None:
+                unmounted_disks.append(disk)
+
+        return missing_disks, unmounted_disks
+
+    def check(self, check_strategy):
+        """
+        Perform additional checks for SnapshotBackupExecutor, specifically:
+
+          - check that the VM instance for which snapshots should be taken exists
+          - check that the expected disks are attached to that instance
+          - check that the attached disks are mounted on the filesystem
+
+        :param CheckStrategy check_strategy: the strategy for the management
+             of the results of the various checks
+        """
+        super(SnapshotBackupExecutor, self).check(check_strategy)
+        if self.server.config.disabled:
+            # Skip checks if the server is not active
+            return
+        check_strategy.init_check("snapshot instance exists in zone")
+        if not self.snapshot_interface.instance_exists(
+            self.snapshot_instance, self.snapshot_zone
+        ):
+            check_strategy.result(
+                self.config.name,
+                False,
+                hint="cannot find compute instance %s in zone %s"
+                % (self.snapshot_instance, self.snapshot_zone),
+            )
+            return
+        else:
+            check_strategy.result(self.config.name, True)
+
+        check_strategy.init_check("snapshot disks attached to instance")
+        cmd = unix_command_factory(self.config.ssh_command, self.server.path)
+        missing_disks, unmounted_disks = self.find_missing_and_unmounted_disks(
+            cmd,
+            self.snapshot_interface,
+            self.snapshot_instance,
+            self.snapshot_zone,
+            self.snapshot_disks,
+        )
+
+        if len(missing_disks) > 0:
+            check_strategy.result(
+                self.config.name,
+                False,
+                hint="cannot find snapshot disks attached to instance %s: %s"
```

Comment:
This reads a bit weird because "snapshots disks" sounds like the disks should already be cloned from snapshots, but what we really mean is that they are disks for which snapshots should be taken at backup time.

Reply:
On reflection I think this is ok since `snapshot_disks` is language already used in the Barman config / barman-cloud args.